main/test/ca/uwaterloo/flix/language/feature/Test.Expression.Lambda.flix

```diff
@@ -0,0 +1,139 @@
+/////////////////////////////////////////////////////////////////////////////
+// Lambdas - Expression.{MkClosureRef,ApplyRef,ApplyClosure}               //
+// Note that closure conversion and lambda lifting means we don't actually //
+// have lambdas in the AST. A lot of functionality is tested indirectly    //
+// by pattern matching.                                                    //
+/////////////////////////////////////////////////////////////////////////////
+
+// TODO: More tests when the typer handles lambda expressions.
+// Test actual lambda expressions (not just top-level definitions): passing them around, free variables, etc.
+// Note: this also applies to comparison (?) of lambdas and using lambdas in let-expressions.
```

Comment:
We should be able to do this now. See also #477 for all the stuff we should test.